components/script/script_thread.rs

```diff
@@ -2017,12 +2018,13 @@ impl ScriptThread {
                     Some(browsing_context) => browsing_context.active_document(),
                     None => return warn!("Message sent to closed pipeline {}.", parent_pipeline_id),
                 };
-                let url = document.url();
+                let url = (*document.url()).clone();
                 if &url[..Position::AfterQuery] == &nurl[..Position::AfterQuery] &&
                     load_data.method == Method::Get {
                     self.check_and_scroll_fragment(fragment, parent_pipeline_id, document.r());
+                    document.set_url(nurl);
```

Comment:
Well, Eventually I think this commit needs to be reverted because a lot of test failed, and I push this commit mostly to notify what actually happened (and to sync between different computers). I hope to get some design suggestions on how the URL fragment should be stored for both the location API and for navigation management.